packages/ui/src/Tabs/Tab.tsx

```diff
@@ -19,22 +19,34 @@ export type ButtonTabProps = {
 
 type TabTypeProps = AnchorTabProps | ButtonTabProps
 
-type TabIconProps =
+type TabLabelProps =
   | {
+      label: string
       icon: IconProps['icon']
       iconAriaLabel: string
+      children?: never
+      ariaLabel?: never
     }
   | {
+      label: string
       icon?: never
       iconAriaLabel?: never
+      children?: never
+      ariaLabel?: never
     }
+  | {
+      label?: never
+      icon?: never
+      iconAriaLabel?: never
+      children: ReactNode
+      ariaLabel: string
+    }
+
 export type TabCommonProps = {
-  label: string
```

Comment:
How about we just make the type of label `ReactNode` instead of creating a third set of TabLabelProps?
WDUT @utkukaratas @Sodik?

Reply:
And by the way - this is our global problem. Almost all of our`label` props are `string` but it's very often the case, that you need to pass something "bigger" than a string, for example a string with a `span` or a string with an Icon.

Reply:
I would say we could even make a separate PR, where we upgrade all of our labels from a `string` to the `ReactNode`

Reply:
I think, that using `label` as `ReactNode` is a not good practice at all. `Label` is a text, for `ReactNode` we have a `children`. 
When we use `label` as `ReactNode`, we can have some problems with other properties, like Icons.
In this case, using `ReactNode` is opposite to using `label` as text + `icon`.

Reply:
In my opinion this:
```
<tab value={0}>Tab <b>1</b></tab>
```
better than that:
```
<tab value={0} label={Tab <b>1</b>}/>
```

Reply:
And I want to point out that this is not just a code problem. This is an important feature in using the library and its root API.

Reply:
In case when we use `label` as `ReactNode` we can do that:
```<tab value={0} label={Tab <b>1</b>} icon={Info} iconAriaLabel="Info" />```
The result will be not predictable at all, I guess. And typing will not help as in this situation.

Reply:
1. If the client wants to put `Icon` into `icon` and then the same `Icon` as a part of the label it's his problem.
2. I think it's pretty predictable - there will be 2 `Icon`s :)

Reply:
And the link with a discussion like this 
https://stackoverflow.com/questions/56003891/react-props-vs-children-what-to-use-when